device-registry/src/main/scala/org/genivi/sota/device_registry/daemon/DeviceUpdateStatusListener.scala

```diff
@@ -0,0 +1,28 @@
+package org.genivi.sota.device_registry.daemon
+
+import java.time.Instant
+
+import org.genivi.sota.data.UpdateStatus
+import org.genivi.sota.device_registry.db.DeviceRepository
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages.{DeviceUpdateStatus, UpdateSpec}
+import slick.driver.MySQLDriver.api._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object DeviceUpdateStatusListener {
+
+  def action(messageBus: MessageBusPublisher)
+            (implicit db: Database, system: ExecutionContext): UpdateSpec => Future[Unit] = {
+    import org.genivi.sota.data.DeviceSearchCommon._
+    (msg: UpdateSpec) =>
+      db.run(for {
+        device <- DeviceRepository.findByUuid(msg.device)
+        status = currentDeviceStatus(device.lastSeen, Seq((Instant.now(), UpdateStatus.withName(msg.status))))
+        _      <- DeviceRepository.setDeviceStatus(device.uuid, status)
+      } yield (device, status)).flatMap { case (device, status) =>
+        messageBus.publish(DeviceUpdateStatus(device.namespace, device.uuid, status))
+      }
```

Comment:
This could be:

```
     db.run { for {
       device <- DeviceRepository.findByUuid(msg.device)
       status = currentDeviceStatus(device.lastSeen, Seq((Instant.now(),UpdateStatus.withName(msg.status))))
       _  <- DeviceRepository.setDeviceStatus(device.uuid,status)
       _ <- messageBus.publish(DeviceUpdateStatus(device.namespace,device.uuid,status))
     } yield ()
    }
```